docs/src/appInsights.ts

```diff
@@ -0,0 +1,32 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// eslint-disable-next-line import/no-unresolved
+import siteConfig from "@generated/docusaurus.config";
+import { ReactPlugin } from "@microsoft/applicationinsights-react-js";
+import { ApplicationInsights } from "@microsoft/applicationinsights-web";
+
+const reactPlugin = new ReactPlugin();
+let appInsights: ApplicationInsights | undefined;
+
+if (siteConfig?.customFields?.INSTRUMENTATION_KEY === undefined) {
+	console.warn("Instrumentation Key is missing. App Insights will not be initialized.");
+}
+
+// Only initialize Application Insights if not in local development.
+// Remove the condition if you want to run Application Insights locally.
+if (typeof window !== "undefined" && window.location.hostname !== "localhost") {
```

Comment:
I wonder if we should normalize around `process.env.NODE_ENV` checks ("production" vs "development") instead of more ad-hoc checks like this.

Docusaurus uses Webpack under the hood, and these variables _do_ get injected. Though it looks like (at least by default) `docusaurus start` sets it to "development" and `docusaurus build` sets it to "production".
https://github.com/facebook/docusaurus/discussions/7621

Since we use build + serve flow for local testing of various things (e.g., search), we would need to override that default behavior for local builds. I'm assuming the behavior can be overriden via CLI arguments, though I haven't checked. Presumably we could add separate dev and production build scripts, and _only_ use the prod build script in the publish pipeline.

Reply:
If we want to follow up in a future PR, let's at least add a TODO here explaining our plans.